examples: streaming example is not runnable (missing package.json/deps, then crashes because createSurf() is not awaited)

## Summary
The `examples/streaming` example is currently not runnable as documented.

There are two separate problems:
1. The example directory does not include its own `package.json`, so `node server.js` fails immediately on a clean checkout because `express` is not installed in that directory.
2. After installing the missing deps locally, the server still crashes because it calls async `createSurf()` without `await`, just like the Express example.

## Reproduction
Environment used:
- macOS Darwin 25.2.0
- Node.js v22.22.0
- repo checkout at current `main` in `hauselabs/surf`

### Repro A: documented commands on clean checkout
`examples/streaming/README.md` says:
```bash
# Terminal 1 — start the server
node server.js

# Terminal 2 — run the streaming client
node client.js
```

From a clean checkout:
```bash
cd examples/streaming
node server.js
```

Actual result:
```txt
Error: Cannot find module 'express'
Require stack:
- .../examples/streaming/server.js
```

### Repro B: after adding the missing local deps
I created a temporary local `examples/streaming/package.json` only to continue validation, with:
- `express`
- `@surfjs/core` via `file:../../packages/core`

Then ran:
```bash
cd examples/streaming
npm install
node server.js
```

Actual result:
```txt
/Users/agentzero/.openclaw/workspace-hausdev/projects/surf/examples/streaming/server.js:66
app.use(surf.middleware());
             ^

TypeError: surf.middleware is not a function
```

## Why this happens
Current `@surfjs/core` exports `async function createSurf(config): Promise<SurfInstance>`, but the example does:
```js
const surf = createSurf({ ... });
app.use(surf.middleware());
```
So `surf` is a Promise at runtime, not a Surf instance.

## Impact
- The streaming example cannot be run as documented.
- SSE behavior cannot be validated from the published example because the server never starts.
- `client.js` is effectively blocked by server startup failure.

## Expected
- The example should include installable dependency metadata (or be wired into the workspace in a documented way).
- The server should await `createSurf()` before mounting middleware.
- After that, the README commands should produce a live SSE stream on `POST /surf/execute` with `{ stream: true }`.

## Notes
I am intentionally not proposing code changes here beyond the issue description; this report is based on live runtime validation.

--- a/SPEC.md
+++ b/SPEC.md
@@ -596,7 +596,7 @@ A **channel** is a string identifier (e.g. `project-123`, `document-abc`) that g
 Channels are opt-in — Surf Live must be explicitly enabled in the server config:
 
 ```typescript
-const surf = createSurf({
+const surf = await createSurf({
   name: 'My App',
   commands: { /* ... */ },
   channels: {

--- a/examples/express/index.js
+++ b/examples/express/index.js
@@ -18,7 +18,8 @@ const carts = new Map(); // sessionId → items[]
 
 // ─── Surf setup ──────────────────────────────────────────────────────────────
 
-const surf = createSurf({
+async function main() {
+  const surf = await createSurf({
   name: 'Example Store',
   description: 'A simple store demonstrating Surf.js with Express',
   version: '1.0.0',
@@ -113,15 +114,21 @@ const surf = createSurf({
       },
     },
   },
-});
+  });
 
-// Mount Surf middleware
-app.use(surf.middleware());
+  // Mount Surf middleware
+  app.use(surf.middleware());
 
-// Start server
-const PORT = process.env.PORT || 3000;
-app.listen(PORT, () => {
-  console.log(`🏄 Store running at http://localhost:${PORT}`);
-  console.log(`   Manifest: http://localhost:${PORT}/.well-known/surf.json`);
-  console.log(`   Execute:  POST http://localhost:${PORT}/surf/execute`);
+  // Start server
+  const PORT = process.env.PORT || 3000;
+  app.listen(PORT, () => {
+    console.log(`🏄 Store running at http://localhost:${PORT}`);
+    console.log(`   Manifest: http://localhost:${PORT}/.well-known/surf.json`);
+    console.log(`   Execute:  POST http://localhost:${PORT}/surf/execute`);
+  });
+}
+
+main().catch((error) => {
+  console.error('Failed to start Express example:', error);
+  process.exit(1);
 });

--- a/examples/fastify/README.md
+++ b/examples/fastify/README.md
@@ -18,7 +18,7 @@ import Fastify from 'fastify'
 import { createSurf } from '@surfjs/core'
 import { fastifyPlugin } from '@surfjs/core/fastify'
 
-const surf = createSurf({ name: 'My API', commands: { ... } })
+const surf = await createSurf({ name: 'My API', commands: { ... } })
 const app = Fastify()
 
 app.register(fastifyPlugin(surf))

--- a/examples/nextjs/README.md
+++ b/examples/nextjs/README.md
@@ -2,13 +2,19 @@
 
 Next.js App Router integration with Surf.js.
 
-## Structure
+## Status
+
+This folder is a **code snippet example**, not a standalone runnable Next.js app.
+It intentionally shows the minimal Surf-specific files you need to add inside an existing Next.js project.
+
+If you want a complete working app, create a Next.js project first and then copy these files into it.
+
+## Included files
 
 ```
-app/api/surf/
-  route.ts           → GET manifest (wire to /.well-known/surf.json via rewrite)
-  execute/route.ts   → POST command execution
-  surf-instance.ts   → Shared Surf instance
+app/api/surf/[...slug]/route.ts    → Surf GET/POST route handler
+app/api/surf/surf-instance.ts      → Shared async Surf instance
+middleware.ts                      → Optional well-known manifest rewrite
 ```
 
 ## Commands
@@ -17,24 +23,32 @@ app/api/surf/
 - `search` — Search by name
 - `addToCart` — Add item to cart
 
-## Next.js Config
+## Use it in a real Next.js app
 
-Add a rewrite to serve the manifest at the standard well-known path:
-
-```ts
-// next.config.ts
-export default {
-  async rewrites() {
-    return [
-      { source: '/.well-known/surf.json', destination: '/api/surf' },
-    ];
-  },
-};
-```
-
-## Run
+1. Create a Next.js app:
 
 ```bash
-npm install
-npm run dev
+npx create-next-app@latest my-surf-app
+cd my-surf-app
+npm install @surfjs/core @surfjs/next
 ```
+
+2. Copy the example files from this folder into your app.
+
+3. Make sure your catch-all route exports the handlers created by `createSurfRouteHandler(surf)`.
+
+## Well-known discovery
+
+Add a rewrite so agents can find the manifest at the standard path:
+
+```ts
+// middleware.ts
+import { surfMiddleware } from '@surfjs/next/middleware';
+
+export default surfMiddleware();
+export const config = { matcher: ['/.well-known/surf.json'] };
+```
+
+## Note on async setup
+
+`createSurf()` is async, so the shared instance should be created with `await createSurf(...)`.

--- a/examples/nextjs/app/api/surf/surf-instance.ts
+++ b/examples/nextjs/app/api/surf/surf-instance.ts
@@ -12,7 +12,7 @@ const cart: Array<{ sku: string; name: string; price: number; quantity: number }
 
 // ─── Surf instance ───────────────────────────────────────────────────────────
 
-export const surf = createSurf({
+export const surf = await createSurf({
   name: 'Next.js Store',
   description: 'Example Next.js integration with Surf.js',
   version: '1.0.0',

--- a/examples/security-headers/index.ts
+++ b/examples/security-headers/index.ts
@@ -140,7 +140,7 @@ function securityHeaders(options: SecurityHeadersOptions = {}): SurfMiddleware {
 
 // ─── Example Usage ───────────────────────────────────────────────────────────
 
-const surf = createSurf({
+const surf = await createSurf({
   name: 'Secure API',
   description: 'Example API with security headers middleware',
   version: '1.0.0',

--- a/examples/streaming/README.md
+++ b/examples/streaming/README.md
@@ -5,11 +5,14 @@ Server-Sent Events (SSE) streaming with Surf.js.
 ## Run
 
 ```bash
+cd examples/streaming
+npm install
+
 # Terminal 1 — start the server
-node server.js
+npm start
 
 # Terminal 2 — run the streaming client
-node client.js
+npm run client
 ```
 
 ## Commands

--- a/examples/streaming/package.json
+++ b/examples/streaming/package.json
@@ -1,0 +1,15 @@
+{
+  "name": "surf-streaming-example",
+  "version": "1.0.0",
+  "private": true,
+  "description": "Surf.js SSE streaming example",
+  "main": "server.js",
+  "scripts": {
+    "start": "node server.js",
+    "client": "node client.js"
+  },
+  "dependencies": {
+    "@surfjs/core": "file:../../packages/core",
+    "express": "^4.22.1"
+  }
+}

--- a/examples/streaming/server.js
+++ b/examples/streaming/server.js
@@ -4,7 +4,8 @@ const { createSurf } = require('@surfjs/core');
 const app = express();
 app.use(express.json());
 
-const surf = createSurf({
+async function main() {
+  const surf = await createSurf({
   name: 'Streaming Example',
   description: 'Demonstrates SSE streaming with Surf.js',
   version: '1.0.0',
@@ -61,11 +62,17 @@ const surf = createSurf({
       },
     },
   },
-});
+  });
 
-app.use(surf.middleware());
+  app.use(surf.middleware());
 
-const PORT = process.env.PORT || 3001;
-app.listen(PORT, () => {
-  console.log(`🏄 Streaming server at http://localhost:${PORT}`);
+  const PORT = process.env.PORT || 3001;
+  app.listen(PORT, () => {
+    console.log(`🏄 Streaming server at http://localhost:${PORT}`);
+  });
+}
+
+main().catch((error) => {
+  console.error('Failed to start streaming example:', error);
+  process.exit(1);
 });

--- a/packages/core/src/adapters/hono.ts
+++ b/packages/core/src/adapters/hono.ts
@@ -291,7 +291,7 @@ function buildHonoApp(surf: SurfInstance, Hono: HonoConstructor): HonoApp {
  * import { createSurf } from '@surfjs/core'
  * import { honoApp } from '@surfjs/core/hono'
  *
- * const surf = createSurf({ ... })
+ * const surf = await createSurf({ ... })
  * const app = new Hono()
  * app.route('/', await honoApp(surf))
  * ```
@@ -318,7 +318,7 @@ export async function honoApp(surf: SurfInstance): Promise<HonoApp> {
  * import { createSurf } from '@surfjs/core'
  * import { honoAppSync } from '@surfjs/core/hono'
  *
- * const surf = createSurf({ ... })
+ * const surf = await createSurf({ ... })
  * const app = new Hono()
  * app.route('/', honoAppSync(surf, Hono))
  * ```

--- a/packages/core/src/namespace.ts
+++ b/packages/core/src/namespace.ts
@@ -55,7 +55,7 @@ export function isCommandDefinition(value: unknown): value is CommandDefinition 
  * @example
  * ```ts
  * // Spread with prefix:
- * const surf = createSurf({
+ * const surf = await createSurf({
  *   commands: {
  *     ...group('cart', {
  *       add: { description: 'Add to cart', run: async (p) => {} },

--- a/packages/devui/src/server.ts
+++ b/packages/devui/src/server.ts
@@ -30,7 +30,7 @@ export interface DevUI {
  * import { createSurf } from '@surfjs/core';
  * import { createDevUI } from '@surfjs/devui';
  *
- * const surf = createSurf({ name: 'My App', commands: { ... } });
+ * const surf = await createSurf({ name: 'My App', commands: { ... } });
  * const devui = createDevUI(surf, { port: 4242 });
  *
  * const { url } = await devui.start();

--- a/packages/next/src/index.ts
+++ b/packages/next/src/index.ts
@@ -10,7 +10,7 @@
  * import { createSurf } from '@surfjs/core';
  * import { createSurfRouteHandler } from '@surfjs/next';
  *
- * const surf = createSurf({ name: 'my-app', commands: { ... } });
+ * const surf = await createSurf({ name: 'my-app', commands: { ... } });
  * export const { GET, POST } = createSurfRouteHandler(surf);
  * ```
  */
@@ -47,7 +47,7 @@ type NextRequest = Request & { nextUrl?: URL };
  * import { createSurf } from '@surfjs/core';
  * import { createSurfRouteHandler } from '@surfjs/next';
  *
- * const surf = createSurf({ name: 'my-app', commands: { ... } });
+ * const surf = await createSurf({ name: 'my-app', commands: { ... } });
  * export const { GET, POST } = createSurfRouteHandler(surf);
  * ```
  *

--- a/packages/next/src/pages.ts
+++ b/packages/next/src/pages.ts
@@ -55,7 +55,7 @@ function headerValue(val: string | string[] | undefined): string | undefined {
  * import { createSurf } from '@surfjs/core';
  * import { createSurfApiHandler } from '@surfjs/next/pages';
  *
- * const surf = createSurf({ name: 'my-app', commands: { ... } });
+ * const surf = await createSurf({ name: 'my-app', commands: { ... } });
  * export default createSurfApiHandler(surf);
  * ```
  *

